tests/test_client.py

```diff
@@ -257,34 +278,72 @@ def test_creds_refreshed(
         datetime.timezone.utc
     )
 
+def test_refresh_creds_func(
+    mocker: MockerFixture, set_env_vars, auth_client_mock, unexpired_auth_credentials_mock, expired_auth_credentials_mock, aws_stubber
+):
+    """Test that auth credentials are refreshed when expired"""
+
+    # mocker.patch("boto3.client", return_value=client)
+    mocker.patch(
+        "msal.PublicClientApplication",
+        return_value=auth_client_mock,
+    )
+
+    client = PrescientClient()
+
+    client._auth_credentials = unexpired_auth_credentials_mock
+
+    # check that when the auth_creds are used they get refreshed from the mock fixture
+    assert client.auth_credentials["id_token"] == "cached_token"
+
+    assert not client.credentials_expired
+
+    with aws_stubber:
+        client.refresh_credentials()
+
+    assert client.auth_credentials["id_token"] == "cached_token"
```

Comment:
just to confirm, the intent is that in this case a) with still valid, cached credentials b) without the use of the `Force=True` arg that there should be no change to the credentials (hence `id_token == "cached_token"`)

Reply:
Correct. Because the creds are still active. I guess I need to add one for when they are expired.